geo/src/types.rs

```diff
@@ -57,38 +57,7 @@ impl<F: GeoFloat> Closest<F> {
 ///     fn foo_1(&self, coord: Coordinate<T>)  -> bool { true }
 ///     fn foo_2(&self)  -> i32 { 9 }
 /// }
-/// impl<T: GeoNum> Foo<T> for Line<T> {
-///     fn foo_1(&self, coord: Coordinate<T>)  -> bool { true }
-///     fn foo_2(&self)  -> i32 { 8 }
-/// }
-/// impl<T: GeoNum> Foo<T> for Rect<T> {
-///     fn foo_1(&self, coord: Coordinate<T>)  -> bool { true }
-///     fn foo_2(&self)  -> i32 { 7 }
-/// }
-/// impl<T: GeoNum> Foo<T> for Triangle<T> {
-///     fn foo_1(&self, coord: Coordinate<T>)  -> bool { true }
-///     fn foo_2(&self)  -> i32 { 6 }
-/// }
-/// impl<T: GeoNum> Foo<T> for Polygon<T> {
-///     fn foo_1(&self, coord: Coordinate<T>)  -> bool { true }
-///     fn foo_2(&self)  -> i32 { 5 }
-/// }
-/// impl<T: GeoNum> Foo<T> for MultiPoint<T> {
-///     fn foo_1(&self, coord: Coordinate<T>)  -> bool { false }
-///     fn foo_2(&self)  -> i32 { 4 }
-/// }
-/// impl<T: GeoNum> Foo<T> for MultiPolygon<T> {
-///     fn foo_1(&self, coord: Coordinate<T>)  -> bool { false }
-///     fn foo_2(&self)  -> i32 { 3 }
-/// }
-/// impl<T: GeoNum> Foo<T> for GeometryCollection<T> {
-///     fn foo_1(&self, coord: Coordinate<T>)  -> bool { false }
-///     fn foo_2(&self)  -> i32 { 2 }
-/// }
-/// impl<T: GeoNum> Foo<T> for MultiLineString<T> {
-///     fn foo_1(&self, coord: Coordinate<T>)  -> bool { false }
-///     fn foo_2(&self)  -> i32 { 1 }
-/// }
+/// ... and so on
```

Comment:
Good comments about these macros! They were kind of a mess. I've done some juggling which I think addresses all your feedback:

- I wanted an example of how to use it in the docs, so I fleshed out the entire example, and removed the `ignore`.
- This revealed a bug... I need to prefix the inner helper macro with `$crate::` not `crate::`.
- I also moved the doc comment from this helper method to the public `geometry_delegate_impl!` macro.
- I prefixed the helper method with `__` to indicate that it's intended to be called directly.
- I marked `geometry_cow_delegate_impl` as `doc(hidden)`. Since `GeometryCow` is `pub(crate)` I want `geometry_cow_delegate_impl` to have similarly restricted visibility, but I don't know enough about macros to know if there's a better way than just hiding it from the docs. Is there a better way?